src/components/PostList.tsx

```diff
@@ -0,0 +1,116 @@
+
+import React, { ChangeEvent, useEffect } from 'react';
+import { getAllData } from '../helpers/api';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+import { UserType, PostType, CommentType } from '../types';
+import { SET_COMMENTS } from '../store/comments';
+import { SET_USERS } from '../store/users';
+import { SET_POSTS } from '../store/posts';
+import { RootState } from '../store';
+import { SET_MESSAGE } from '../store/message';
+import { Post } from './Post';
+import { setFilterFieldCreator } from '../store/filterField';
+import { setFilteredPostsCreator } from '../store/filteredPosts';
+import { debounce } from '../helpers/debounce';
+
+type PropsType = {
+  setMessage: (message: string) => void;
+  setUsers: (users: UserType[]) => void;
+  setComments: (comments: CommentType[]) => void;
+  setPosts: (posts: PostType[]) => void;
+  message: string;
+  posts: PostType[];
+  filterField: string;
+  setFilterField: (filterField: string) => void;
+  setFilteredPosts: (filteredPosts: PostType[]) => void;
+  filteredPosts: PostType[],
```

Comment:
don't store two arrays with the same posts in the state, state knows about the initial array only